slimta/relay/smtp/client.py

```diff
@@ -186,12 +186,18 @@ def _deliver(self, result, envelope):
         try:
             self._handle_encoding(envelope)
             self._send_envelope(rcpt_results, envelope)
-            self._send_message_data(envelope)
+            msg_result = self._send_message_data(envelope)
         except SmtpRelayError as e:
             result.set_exception(e)
             self._rset()
         else:
-            result.set(rcpt_results)
+            if rcpt_results is None:
```

Comment:
I don't believe `rcpt_results` can ever be `None`, as it is initialized to an array at the top of this function and never re-initialized. Perhaps something like this would work better:

```python
if [res for res in rcpt_results if res is not None]:
```